### DISCUSSION TITLE
Animated Update Of Dom Node

### DISCUSSION THREAD

**Discussion by tommillard** [lit/lit#2865] [Q&A]
One of the elements produced my my render function is a bunch of simple counters in <span> tags. I was wondering if anyone could suggest the best way to apply an animation to any of the counters if their displayed value changes? I'm thinking something like transforming its scale from 1 > 1.5 > 1 so it bulges when it updates.

Some options I've considered:

- LitLabs Motion. Love this for other types of animation, but I can't seem to find a way to animate something that isn't moving from one place to another.
- Hooking in to willUpdate(). I could check each individual value here and add an extra property to the values that are changing, then reference that in the render function to add a class or attribute. I think this would work, but it seems messy.
- Extracting the counter into a separate LitElement. This way the willUpdate() function could apply a class and remove it again after a timeout. Then I could hook a CSS animation onto the presence of that class.

I like the last option the best, but wondered if there was a built-in directive I'm missing, or some other technique?

Thanks,

**Comment by e111077** [comment]
If you ever need to step out of the normal rendering lifecycle for a small part, consider using an `AsyncDirective` The `update` method of `AsyncDirective` gives you access to the node itself and you can apply the [Web Animations API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API). We have a tutorial on async directives here:

https://lit.dev/tutorials/async-directive/

**Comment by sorvell** [comment]
It would be nice to have a simple, declarative way to use the Web Animations API. Here's a quick example of what this might look like. There's a simple directive that's just being used to apply the given animation parameters. Was thinking that something like this might make sense in the motion package.

https://lit.dev/playground/?mods=gists#gist=3fd852836147125354fba72a5ef81227

**Comment by kevinpschaaf** [comment]
Here's an example of a simple directive that makes the parent node bulge when the value rendered into it changes: 

https://lit.dev/playground/#gist=3eddbc798396cd4c26ca6cbbc491a8b6
